### PR TITLE
Added functionality for modifing graph metadata using UI

### LINKED ISSUE #317: Modify graph metadata using UI
The main missing UI features to modify graph metadata are:
- [ ] Modify graph title
- [ ] Add/remove tags
- [ ] Modify Graph Information / Legend

### PR DESCRIPTION
**Pull request by bruce-wayne99**
Fixes #317 
### Implemented UI for modifying user meta-data which includes the following
- User will be able to modify graph attributes by clicking on the "edit" button.
- User will be able to modify description in "Graph Information" tab.
- User will be able to modify tags, title, name in the edit panel which appears on clicking "edit" button.
- Corresponding changes will also be reflected in the database.
#### Gif  of the implementation
pull_request

### CODE REVIEW

**Comment by adbharadwaj** [comment]
@bruce-wayne99 Can you add documentation for the feature as well. You can find the markdown documentation [here](https://github.com/Murali-group/GraphSpace/tree/master/docs).

**Comment by tmmurali** [comment]
@adbharadwaj it might help to provide @bruce-wayne99 more information on what you expect from these docs and where he should add them in the docs.

**Comment by bruce-wayne99** [comment]
@adbharadwaj , @tmmurali  Sir,
I have added documentation with gifs for this feature. Please review

File: applications/graphs/dal.py [hunk 1]
```diff
@@ -290,6 +290,9 @@ def add_tag(db_session, name):
 	db_session.add(tag)
 	return tag
 
+@with_session
+def remove_tags_by_graph_id(db_session, graph_id):
```
Comment (adbharadwaj) [inline applications/graphs/dal.py]: @bruce-wayne99 Why was this method introduced?

File: applications/graphs/controllers.py [hunk 2]
```diff
@@ -229,6 +229,11 @@ def update_graph(request, graph_id, name=None, is_public=None, graph_json=None,
 		if name is not None:
 			G.set_name(name)
 
+		db.remove_tags_by_graph_id(request.db_session, graph_id=graph_id)
```
Comment (adbharadwaj) [inline applications/graphs/controllers.py]: @bruce-wayne99 Why was this method introduced?

File: static/js/graphs_page.js [hunk 3]
```diff
@@ -430,6 +430,7 @@ var uploadGraphPage = {
 var graphPage = {
     cyGraph: undefined,
     timeout: null,
+    tagsBar: null,
```
Comment (adbharadwaj) [inline static/js/graphs_page.js]: Rename it to `tagsEditor`

File: static/js/graphs_page.js [hunk 4]
```diff
@@ -686,6 +700,56 @@ var graphPage = {
...
+    },
+    graphEditFormatter: function(e, title, name, description) {
+
+        //populating input tags with appropriate values before showing the edit panel.
+        $('#new_name').val(name);
+        $('#new_title').val(title);
+        $('#new_description').val(description);
+
+        //fetching all the tags and populating them in the tag bar.
+        console.log(title, name, description);
```
Comment (adbharadwaj) [inline static/js/graphs_page.js]: Remove logs from final pull request.

File: applications/graphs/controllers.py (hunk 2 above)
Comment (bruce-wayne99) [reply applications/graphs/controllers.py]: Editing tags involves **adding new tags** and **removing old tags** simultaneously, 
For example:
Let a user removed a tag named **x** from the graph **G** and added a tag **y** to the graph. So, there is a need to delete a row from `GraphToTag` containing the `graph_id` equal to the graph id of **G** and `tag_id` equal to tag id of tag **x** and also addition of a row with `graph_id` equal to the graph id of **G** and `tag_id` equal to tag id of tag **y**.
Now both these operations can be carried out at once if we remove all the entries of graph **G** in `GraphToTag` table  and after that adding the final list of tags the user wants to the graph.

Comment (bruce-wayne99) [reply applications/graphs/controllers.py]: @adbharadwaj  I have provided an explanation for the usage of `db.remove_tags_by_graph_id`. Please review.

**Comment by bruce-wayne99** [comment]
@adbharadwaj I have provided explanation for a adding the function `db.remove_tags_by_graph_id`. Please review.

**Comment by jddanna** [comment]
Thanks for your contributions! I made a new Pull request #456 with a few minor changes to merge with master. It's easier this way to see the merges with updated code since the branch for this PR was deleted.

**Pull request closed without merging**